Adds zipfile import-db and import-files support, fixes #147

rfay opened the pull request:
## The Problem:

We need to add zipfile support for files and db imports

## The Fix:

* Add the utility to unzip
* Add utility test
* Add capability in import-files and import-db
* Add zipball testing for import-files and import-db

## The Test:

* Import files with a zipball instead of a tarball
* Import database with a zipball instead of a tarball

## Automation Overview:
<!-- Please provide an overview of tests introduced by this PR, or an explanation for why no tests are needed. -->

## Related Issue Link(s):
OP to add zip support is in #147 

## Release/Deployment notes:
<!-- Does this affect anything else, or are there ramifications for other code? Does anything have to be done on deployment? -->

rfay commented:
I think this is ready for review now.

tannerjfco commented:
The panic on straight .sql files is reproducible on Mac as well, so it looks like we have 2 issues in play here;

- importing a zipped database on Windows fails to find a SQL file
- importing an unextracted sql dump results in a panic

We should probably add a test case for importing a sql dump directly.

rfay commented:
* I was able to recreate the stack trace, as it was due to a bug I introduced in this PR. Fixed it.
* Added to the import test to import dbs from all types of files and archives.
* I was unable to recreate the Windows "No file to import found" problem, but have to confess I rebased to include the various Windows path fixes you did @tannerjfco .

Anyway, I think this is ready for another look.
